### ISSUE TITLE
Broken card if one of the questions composing it brought empty result

### ISSUE THREAD

**Issue by Coqueiro** [metabase/metabase#4630]
### Bug

- Browser: Chrome 56.0
- OS: Windows 8
- Database: SQL Server
- Metabase version: v.0.22.2
- Metabase hosting environment: Elastic Beanstalk
- Metabase internal database: Postgres

If a card composed of more than one question has one of these question returning empty results, the card breaks showing a exclamation point image and the warning: "There was a problem displaying this chart."

Also, the question disappear from the card's label (which means I can't disable it, having to create a new card without the empty result query. Sometimes those time series queries **will** bring empty results, and it's ok. If that happens, I, as an user, would want to see it in the card (as a label), but without presence in the graph.

**Comment by mazameli** [comment]
Confirmed this bug, but with a few differences:
- I am not seeing this behavior: "the question disappear from the card's label"
- I did not see this behavior for combined scalar questions; only ones with a time or category breakout
- the warning message reads: `Cannot read property 'slice' of undefined`. The JS console error displays:
```
CardRenderer.jsx?b429:76 TypeError: Cannot read property 'slice' of undefined
    at http://localhost:8080/app/dist/app-main.hot.bundle.js?a84ce1f4fd770a8a94fa:135310:28
    at Array.map (native)
    at lineAreaBar (http://localhost:8080/app/dist/app-main.hot.bundle.js?a84ce1f4fd770a8a94fa:135306:25)
    at Object.barRenderer [as renderer] (http://localhost:8080/app/dist/app-main.hot.bundle.js?a84ce1f4fd770a8a94fa:135532:13)
    at CardRenderer.renderChart (http://localhost:8080/app/dist/app-main.hot.bundle.js?a84ce1f4fd770a8a94fa:122562:43)
    at CardRenderer.componentDidUpdate (http://localhost:8080/app/dist/app-main.hot.bundle.js?a84ce1f4fd770a8a94fa:122523:19)
    at measureLifeCyclePerf (http://localhost:8080/app/dist/app-main.hot.bundle.js?a84ce1f4fd770a8a94fa:33196:13)
    at http://localhost:8080/app/dist/app-main.hot.bundle.js?a84ce1f4fd770a8a94fa:33850:12
    at CallbackQueue.notifyAll (http://localhost:8080/app/dist/app-main.hot.bundle.js?a84ce1f4fd770a8a94fa:25635:23)
    at ReactReconcileTransaction.close (http://localhost:8080/app/dist/app-main.hot.bundle.js?a84ce1f4fd770a8a94fa:36617:27)
```

Steps to repro: 
- Create two saved questions, both broken out by the same dimension, either by time or by a category, displayed as a line, bar, or area chart. One of the cards should produce results, the other should not. (The way I created a card which produced no results was by adding a greater-than filter on a `subtotal` field that was outside the range of present values.)
- Create a new dashboard, add the first question, then add the second one to the first card.
- You should immediately see `Cannot read property 'slice' of undefined` in the preview modal, as well as on the card after you confirm.

<img width="756" alt="screen shot 2017-04-03 at 12 58 14 pm" src="https://cloud.githubusercontent.com/assets/2223916/24629748/fc509e54-186e-11e7-881c-29a705113cbf.png">



OS: macOS Sierra
Browser: Chrome v57
DB: H2 (sample dataset)

**Comment by attekei** [comment]
A similar problem which I will fix now too is that if the original question produces empty result, then "No results!" is shown even if other(s) have results:

![screen shot 2017-04-14 at 12 24 07](https://cloud.githubusercontent.com/assets/6034928/25053644/4a50a774-210d-11e7-9632-225f83997d5b.png)

**Comment by attekei** [comment]
Fixed in #4791
